reactivesocket-cpp/test/tcp/TcpServer.cpp

```diff
@@ -0,0 +1,141 @@
+using namespace ::testing;
+using namespace ::reactivesocket;
+using namespace ::folly;
+
+DEFINE_string(address, "9898", "host:port to listen to");
+
+namespace {
+    class ServerSubscription
+        : public virtual IntrusiveDeleter,
+          public Subscription {
+    public:
+        ~ServerSubscription() {
+        }
+
+        // Subscription methods
+        void request(size_t n) override {
+          // TODO delay sending responses until this is triggered
+        }
+
+        void cancel() override {
+        }
+    };
+
+    class ServerRequestHandler
+        : public RequestHandler {
+    public:
+        /// Handles a new Channel requested by the other end.
+        ///
+        /// Modelled after Producer::subscribe, hence must synchronously call
+        /// Subscriber::onSubscribe, and provide a valid Subscription.
+        Subscriber<Payload> &
+            handleRequestChannel(
+            reactivesocket::Payload request,
+            Subscriber<Payload> &
+            response) override {
+          LOG(ERROR) << "not expecting server call";
+          response.onError(std::runtime_error("incoming channel not supported"));
+
+          auto* subscription = new MemoryMixin<CancelSubscription>();
+          response.onSubscribe(*subscription);
+
+          return *(new MemoryMixin<CancelSubscriber>());
+        }
+
+        /// Handles a new inbound Subscription requested by the other end.
+        void handleRequestSubscription(
+            Payload request,
+            Subscriber<Payload> &
+            response) override {
+
+          auto* subscription = new MemoryMixin<ServerSubscription>();
+          response.onSubscribe(*subscription);
+
+          std::cout << "ServerRequestHandler.handleRequestSubscription " << request->moveToFbString() << "\n";
+
+          response.onNext(folly::IOBuf::copyBuffer("from server"));
+        }
+    };
+
+    class Callback : public AsyncServerSocket::AcceptCallback {
+    public:
+        Callback(EventBase &eventBase) : eventBase_(eventBase) { };
+
+        virtual ~Callback() = default;
+
+        virtual void connectionAccepted(int fd, const SocketAddress &clientAddr) noexcept override {
+          std::cout << "connectionAccepted" << clientAddr.describe() << "\n";
+
+          auto socket = folly::AsyncSocket::UniquePtr(new AsyncSocket(&eventBase_, fd));
+
+          std::unique_ptr<DuplexConnection> connection = folly::make_unique<TcpDuplexConnection>(std::move(socket));
+          std::unique_ptr<DuplexConnection> framedConnection = folly::make_unique<FramedDuplexConnection>(std::move(connection));
+          std::unique_ptr<RequestHandler> requestHandler = folly::make_unique<ServerRequestHandler>();
+
+          reactiveSocket_ = ReactiveSocket::fromClientConnection(std::move(framedConnection), std::move(requestHandler));
+        }
+
+        virtual void acceptError(const std::exception &ex) noexcept override {
+          std::cout << "acceptError" << ex.what() << "\n";
```

Comment:
in case of the error, do you want to terminate the event loop?
At least the program will terminate..